### PR TITLE
Support immutable cache headers for _astro assets

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Adds an immutable cache header for assets inside of `_astro` in the Node adapter.
- Closes https://github.com/withastro/astro/issues/9106

## Testing

- Test case added
- Test to ensure that only stuff inside `_astro` is given the header.

## Docs

N/A, bug fix

### CODE REVIEW

File: .changeset/twelve-fishes-fail.md
Comment (natemoo-re) [inline .changeset/twelve-fishes-fail.md:5]:
```suggestion
Automatically set immutable cache headers for assets served from the `/_astro` directory
```

**Review by natemoo-re — APPROVED** [review]
Seems great!

File: packages/integrations/node/src/http-server.ts
Comment (bluwy) [inline packages/integrations/node/src/http-server.ts:68]: Perhaps we can also align the value as Vercel's  https://github.com/withastro/astro/blob/e3dce215a5ea06bcff1b21027e5613e6518c69d4/packages/integrations/vercel/src/static/adapter.ts#L128

**Review by bluwy — APPROVED** [review]

File: packages/integrations/node/src/http-server.ts
Comment (matthewp) [inline packages/integrations/node/src/http-server.ts:68]:
```suggestion
					_res.setHeader('Cache-Control', 'public, max-age=31536000, immutable')
```

**Comment by matthewp** [comment]
@florian-lefebvre it can be yes, the Vercel adapter already has it.

**Comment by florian-lefebvre** [comment]
Oh ok, looking at the issue I assumed it was concerning every adapter. It may be worth saying it on adapters docs, because I suspect many people (me included) add those headers to vercel/netlify config but if it's done by default, that's even better!

**Comment by matthewp** [comment]
@florian-lefebvre Yeah, Astro core isn't responsible for serving static assets. I agree on the docs, I'll add that to this PR.

**Comment by matthewp** [comment]
Waiting on docs review before merging.

File: packages/integrations/node/README.md
Comment (sarah11918) [inline packages/integrations/node/README.md:195]:
```suggestion
In standalone mode, assets in your `dist/client/` folder are served via the standalone server. You might be deploying these assets to a CDN, in which case the server will never actually be serving them. But in some cases, such as intranet sites, it's fine to serve static assets directly from the application server.
```

File: .changeset/twelve-fishes-fail.md
Comment (sarah11918) [inline .changeset/twelve-fishes-fail.md:5]:
```suggestion
Automatically sets immutable cache headers for assets served from the `/_astro` directory.
```

**Pull request merged**
